portfolio/src/main/webapp/comments.js

```diff
@@ -12,10 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-window.onload = loadComments();
-
 function loadComments(){
-  fetch('/data').then(response => response.json()).then((comments) => {
+  const maxComments = document.getElementById("max-comments-select").value;
```

Comment:
It might be best to cast this to integer to ensure that you're not constructing something weird below when you're concatenating "/data" and "?max_comments".